src/shared/components/UploadProgressBar/UploadProgressBar.tsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react'
+import { CSSTransition, SwitchTransition } from 'react-transition-group'
+
+import { AssetUploadStatus } from '@/providers/uploadsManager/types'
+import { transitions } from '@/shared/theme'
+
+import {
+  BottomProgressBar,
+  ProgressBar,
+  StyledLoader,
+  StyledSvgAlertSuccess,
+  UploadProgressBarContainer,
+} from './UploadProgressBar.style'
+
+export type UploadProgressBarProps = {
+  progress?: number
+  lastStatus?: AssetUploadStatus
+  withLoadingIndicator?: boolean
+  withCompletedAnimation?: boolean
+  className?: string
+}
+
+export const UploadProgressBar: React.FC<UploadProgressBarProps> = ({
+  progress,
+  lastStatus,
+  className,
+  withLoadingIndicator,
+  withCompletedAnimation,
+}) => {
+  return (
+    <UploadProgressBarContainer className={className}>
+      <ProgressBar
+        runCompletedAnimation={withCompletedAnimation}
+        progress={progress || 0}
+        isProcessing={lastStatus === 'processing'}
+        isCompleted={lastStatus === 'completed'}
+      />
+      {lastStatus !== 'completed' && <BottomProgressBar progress={progress || 0} />}
```

Comment:
Let's add default value for `progress` in props destructuring, since it's used in few places